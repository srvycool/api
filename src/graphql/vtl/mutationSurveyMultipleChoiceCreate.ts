export const request = (tableName: string): string => `
  #set($ctx.stash = {})
  $util.qr($ctx.stash.put("node", $util.autoId()))

  #set($entities = [])
  $util.qr($entities.add($util.dynamodb.toMap({ "node": "$ctx.stash.node", "path": "survey", "id": "$ctx.stash.node", "label": "$ctx.arguments.question", "surveyType": "MULTIPLE_CHOICE" }).M))

  #foreach($answer in $ctx.arguments.answers)
    #set($answerID = $util.autoId())
    $util.qr($entities.add($util.dynamodb.toMap({ "node": "$ctx.stash.node", "path": "answer#$answerID", "id": "$answerID", "label": "$answer", "votes": 0 }).M))
  #end

  {
    "version" : "2018-05-29",
    "operation" : "BatchPutItem",
    "tables" : {
      "${tableName}": $utils.toJson($entities)
    }
  }
`;

export const response = `
  {
    "id": "$ctx.stash.node",
    "label": "$ctx.arguments.question"
  }
`;
