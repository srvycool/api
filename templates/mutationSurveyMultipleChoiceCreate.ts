export const request = (tableName: string) => `
  $util.qr($ctx.stash.put("node", $util.autoId()))

  #set($entities = [])
  $util.qr($entities.add($util.dynamodb.toMapValues({ "node": "$ctx.stash.node", "path": "survey", "id": "$ctx.stash.node", "label": "$ctx.args.question", "surveyType": "MULTIPLE_CHOICE" })))

  #foreach($answer in $ctx.args.answers)
    #set($answerID = $util.autoId())
    $util.qr($entities.add($util.dynamodb.toMapValues({ "node": "$ctx.stash.node", "path": "answer#$answerID", "id": "$answerID", "label": "$answer", "votes": 0 })))
  #end

  {
    "version" : "2018-05-29",
    "operation" : "BatchPutItem",
    "tables" : {
      "${tableName}": $utils.toJson($entities)
    }
  }
`;

export const response = `
  #if ($ctx.error)
    $util.error($ctx.error.message, $ctx.error.type, null, $ctx.result.data.unprocessedKeys)
  #end

  {
    "id": "$ctx.stash.node",
    "label": "$ctx.args.question"
  }
`;
