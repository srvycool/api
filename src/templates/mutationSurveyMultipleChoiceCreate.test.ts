jest.mock('uuid', () => ({ v4: () => 'gg-hh-ii' }));

import Parser, { Context } from 'appsync-template-tester';
import { request, response } from './mutationSurveyMultipleChoiceCreate';

const TABLE_NAME = 'myFancyTable';

describe('templates:mutationSurveyMultipleChoiceCreate', () => {
  describe('request', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request(TABLE_NAME));
      const context = {
        arguments: {
          question: 'Coffee?',
          answers: ['Yes', 'No'],
        },
      } as Context;

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe('response', () => {
    it('should match snapshot', () => {
      const parser = new Parser(response);
      const context = {
        stash: {
          node: 'aa-bb-cc',
        },
        arguments: {
          question: 'Coffee?',
        },
      } as Context;

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});
