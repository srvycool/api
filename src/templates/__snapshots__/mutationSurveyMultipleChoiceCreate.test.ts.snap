// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates:mutationSurveyMultipleChoiceCreate request should match snapshot 1`] = `
Object {
  "operation": "BatchPutItem",
  "tables": Object {
    "myFancyTable": Array [
      Object {
        "id": Object {
          "S": "gg-hh-ii",
        },
        "label": Object {
          "S": "Coffee?",
        },
        "node": Object {
          "S": "gg-hh-ii",
        },
        "path": Object {
          "S": "survey",
        },
        "surveyType": Object {
          "S": "MULTIPLE_CHOICE",
        },
      },
      Object {
        "id": Object {
          "S": "gg-hh-ii",
        },
        "label": Object {
          "S": "Yes",
        },
        "node": Object {
          "S": "gg-hh-ii",
        },
        "path": Object {
          "S": "answer#gg-hh-ii",
        },
        "votes": Object {
          "N": 0,
        },
      },
      Object {
        "id": Object {
          "S": "gg-hh-ii",
        },
        "label": Object {
          "S": "No",
        },
        "node": Object {
          "S": "gg-hh-ii",
        },
        "path": Object {
          "S": "answer#gg-hh-ii",
        },
        "votes": Object {
          "N": 0,
        },
      },
    ],
  },
  "version": "2018-05-29",
}
`;

exports[`templates:mutationSurveyMultipleChoiceCreate response should match snapshot 1`] = `
Object {
  "id": "aa-bb-cc",
  "label": "Coffee?",
}
`;
